nes-input-formatters/src/NativeInputFormatIndexer.cpp

```diff
@@ -0,0 +1,33 @@
+/*
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#include <NativeInputFormatIndexer.hpp>
+
+#include <memory>
+#include <utility>
+
+#include <InputFormatIndexerRegistry.hpp>
+
+namespace NES::InputFormatters
+{
+
+InputFormatIndexerRegistryReturnType
+InputFormatIndexerGeneratedRegistrar::RegisterNativeInputFormatIndexer(InputFormatIndexerRegistryArguments arguments)
+{
+    auto inputFormatter = std::make_unique<NativeInputFormatIndexer<false>>();
```

Comment:
I really do not like these bool templates, can you define an alias for them so that I know what's going on here?